packages/studio-ui/src/AppWithLazyLoading.tsx

```diff
@@ -15,24 +15,26 @@ export default function AppWithLazyLoading() {
   ]);
   const componentsLoaded = loadedCount === totalCount;
   const [appLoaded, setAppLoaded] = useState(false);
+  const [stylesLoaded, setStylesLoaded] = useState(false);
+  const resourcesLoaded = appLoaded && stylesLoaded;
 
   useEffect(() => {
     loadComponents();
-    loadStyling();
+    void Promise.all(loadStyling()).then(() => setStylesLoaded(true));
```

Comment:
If a Component or Styling import fails, the loading screen will be displayed indefinitely. I don't think that's something we need to fix as part of this PR. We need to get some guidance from Product & UX on how to handle this case. The hanging of the loading screen when a Component import failed existed since before this PR.